File: e2e/playwright-python/interactions/steps/navigation_step.py
from playwright.sync_api import Page
from decorators.step_decorator import step


class NavigationStep:
    def __init__(self, page: Page):
        self._page = page

    @step("Navigate to main page")
    def open_main_page(self) -> None:
        self._page.goto("/")

    @step("Refresh page")
    def refresh(self) -> None:
        self._page.reload()

File: e2e/cucumber-playwright-typescript/interactions/steps/NavigationStep.ts
import { Page } from '@playwright/test';
import { step } from '@/decorators/step-decorator';

export class NavigationStep {
  constructor(private readonly page: Page) {}

  @step('Navigate to main page')
  async openMainPage(): Promise<void> {
    await this.page.goto('/');
  }

  @step('Refresh page')
  async refresh(): Promise<void> {
    await this.page.reload();
  }
}
